tests/models/test_default_evaluator.py

```diff
@@ -642,6 +645,45 @@ def test_svm_classifier_evaluation(svm_model_uri, breast_cancer_dataset, baselin
     }
 
 
+def _evaluate_explainer_with_exceptions(model_uri, dataset):
+    with mlflow.start_run():
+        evaluate(
+            model_uri,
+            dataset._constructor_args["data"],
+            model_type="classifier",
+            targets=dataset._constructor_args["targets"],
+            dataset_name=dataset.name,
+            evaluators="default",
+            evaluator_config={
+                "ignore_exceptions": False,
+            },
+        )
+
+
+def test_default_explainer_pandas_df_str_cols(
+    multiclass_logistic_regressor_model_uri, iris_pandas_df_dataset
+):
+    _evaluate_explainer_with_exceptions(
+        multiclass_logistic_regressor_model_uri, iris_pandas_df_dataset
+    )
+
+
+def test_default_explainer_pandas_df_num_cols(
+    multiclass_logistic_regressor_model_uri, iris_pandas_df_num_cols_dataset
+):
+    _evaluate_explainer_with_exceptions(
+        multiclass_logistic_regressor_model_uri, iris_pandas_df_num_cols_dataset
+    )
+
+
+def test_default_explainer_pandas_df_longname_cols(
+    multiclass_logistic_regressor_model_uri, iris_pandas_df_longname_cols_dataset
+):
+    _evaluate_explainer_with_exceptions(
+        multiclass_logistic_regressor_model_uri, iris_pandas_df_longname_cols_dataset
+    )
```

Comment:
<img width="1381" alt="image" src="https://user-images.githubusercontent.com/17039389/188412044-5c80681a-dd30-4f27-97a7-0af4630a7e8a.png">

<img width="1381" alt="image" src="https://user-images.githubusercontent.com/17039389/188412282-3688fbf5-a066-49ef-9568-dd3ab3d03ae1.png">

<img width="1381" alt="image" src="https://user-images.githubusercontent.com/17039389/188412176-5d63d471-87f7-4e7f-9bd8-a280a8b63064.png">

The SHAP plots logged in this test look like this. Is this the expect result?

Reply:
Is `(f_3)` expected?

Reply:
Do we made some change on shap plot rendering recently ? I remember previously I adjust the plot layout well.

Reply:
This line says `For duplicated truncated name, attach "(f_{feature_index})" at the end`, but there is no duplicates in the truncated names.

https://github.com/mlflow/mlflow/blob/5fa68ff6addd439ba0f267b30cd73279379b823d/mlflow/models/evaluation/default_evaluator.py#L566

Reply:
> Do we made some change on shap plot rendering recently ? I remember previously I adjusted the plot layout well.

I don't know.

Reply:
> Is (f_3) expected?

Yes. After name being truncated, conflicts might happen, e.g., 

`longlonglonglongabcdefghijklmnlonglonglonglong` and `longlonglonglongxyzxyzxyzxyzxlonglonglonglong`
after truncation, they both becomes `longlonglonglong...longlonglonglong`

> Looks like the layout has some issue when feature name is long

I will try to fix it.

Reply:
@WeichenXu123 Why do we need to add the suffix when there is no duplicate?

Reply:
@harupy 
Truncation fixing PR : https://github.com/mlflow/mlflow/pull/6705

Reply:
![shap_beeswarm_plot_on_data_diabetes_dataset](https://user-images.githubusercontent.com/19235986/188540177-71ff7358-2240-4de8-8745-4f11620338a7.png)
![shap_feature_importance_plot_on_data_diabetes_dataset](https://user-images.githubusercontent.com/19235986/188540202-a4d102bc-9d52-4a09-ae83-be74b264831a.png)
![shap_summary_plot_on_data_diabetes_dataset](https://user-images.githubusercontent.com/19235986/188540218-542d8b03-fff5-4cbb-8066-62d821e0c4d2.png)

In my test plot rendering works well.
@harupy Could you provide your testing code ?

Reply:
@WeichenXu123 You can run `test_default_explainer_pandas_df_longname_cols`.